tests/integration/aws_s3/test_aws_s3.py

```diff
@@ -203,78 +251,107 @@ def test_using_profile(self, use_datastore_profile):
             os.environ.pop(param)
 
-    def test_directory(self, use_datastore_profile):
+    @pytest.mark.parametrize(
+        "file_format, reader, reader_args",
+        [
+            ("parquet", pd.read_parquet, {}),
+            ("csv", pd.read_csv, {}),
+            ("json", pd.read_json, {"orient": "values"}),
+        ],
+    )
+    def test_as_df(
+        self,
+        use_datastore_profile,
+        file_format: str,
+        reader: callable,
+        reader_args: dict,
+    ):
+        # A more advanced test of the as_df function that includes Parquet, CSV, and JSON files.
         param = self.s3["ds"] if use_datastore_profile else self.s3["s3"]
         for p in credential_params:
             os.environ[p] = config["env"][p]
-        parquet_dir = f"/parquets{uuid.uuid4()}"
-        parquets_url = param["bucket_path"] + parquet_dir
-        #  generate dfs
-        # Define data for the first DataFrame
-        data1 = {"Column1": [1, 2, 3], "Column2": ["A", "B", "C"]}
-
-        # Define data for the second DataFrame
-        data2 = {"Column1": [4, 5, 6], "Column2": ["X", "Y", "Z"]}
-
-        # Create the DataFrames
-        df1 = pd.DataFrame(data1)
-        df2 = pd.DataFrame(data2)
-        with (
-            tempfile.NamedTemporaryFile(suffix=".parquet", delete=True) as temp_file1,
-            tempfile.NamedTemporaryFile(suffix=".parquet", delete=True) as temp_file2,
-        ):
-            # Save DataFrames as Parquet files
-            df1.to_parquet(temp_file1.name, index=False)
-            df2.to_parquet(temp_file2.name, index=False)
-            #  upload
-            dt1 = mlrun.run.get_dataitem(parquets_url + "/df1.parquet")
-            dt2 = mlrun.run.get_dataitem(parquets_url + "/df2.parquet")
-            dt1.upload(src_path=temp_file1.name)
-            dt2.upload(src_path=temp_file2.name)
-            dt1.as_df()
-            dt2.as_df()
-            dt_dir = mlrun.run.get_dataitem(parquets_url)
-            tested_df = dt_dir.as_df(format="parquet")
-            expected_df = pd.concat([df1, df2], ignore_index=True)
-            assert_frame_equal(tested_df, expected_df)
-
-    def test_directory_csv(self, use_datastore_profile):
+        filename = f"/df_{uuid.uuid4()}.{file_format}"
+        file_url = param["bucket_path"] + self.run_dir + filename
+
+        local_file_path = join(self.assets_path, f"test_data.{file_format}")
+        source = reader(local_file_path, **reader_args)
+
+        upload_data_item = mlrun.run.get_dataitem(file_url)
+        upload_data_item.upload(local_file_path)
+        response = upload_data_item.as_df(**reader_args)
+        pd.testing.assert_frame_equal(source, response)
+
+    @pytest.mark.parametrize(
+        "file_format, reader, reader_args",
+        [
+            ("parquet", dd.read_parquet, {}),
+            ("csv", dd.read_csv, {}),
+            ("json", dd.read_json, {"orient": "values"}),
+        ],
+    )
+    def test_as_df_dd(
```

Comment:
This test is a duplicate of `test_as_df`, expect for the `as_df` at the end. Why not just keep the original test and add the dask check at the end? It saves us from
* Duplicating a bunch of tests
* Effectively doubling the size of the parameter matrix

Reply:
tnx for the comment! done.